src/Illuminate/Support/JsString.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Contracts\Support\Htmlable;
+use Illuminate\Contracts\Support\Jsonable;
+use JsonSerializable;
+
+class JsString implements Htmlable
+{
+    /**
+     * Flags that must always be used when encoding to JSON for JsString.
+     *
+     * @var int
+     */
+    protected const REQUIRED_FLAGS = JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR;
+
+    /**
+     * The javascript string.
+     *
+     * @var string
+     */
+    protected $js;
+
+    /**
+     * Create a new JsString from data.
+     *
+     * @param  mixed  $data
+     * @param  int  $flags
+     * @param  int  $depth
+     * @return static
+     *
+     * @throws \JsonException
+     */
+    public static function from($data, $flags = 0, $depth = 512)
+    {
+        return new static($data, $flags, $depth);
+    }
+
+    /**
+     * Create a new JsString.
+     *
+     * @param  mixed  $data
+     * @param  int|null  $flags
+     * @param  int  $depth
+     * @return void
+     *
+     * @throws \JsonException
+     */
+    public function __construct($data, $flags = 0, $depth = 512)
+    {
+        $this->js = $this->convertDataToJavaScriptExpression($data, $flags, $depth);
+    }
+
+    /**
+     * Get string representation of data for use in HTML.
+     *
+     * @return string
+     */
+    public function toHtml()
+    {
+        return $this->js;
+    }
+
+    /**
+     * Get string representation of data for use in HTML.
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->toHtml();
+    }
+
+    /**
+     * Convert data to a JavaScript expression.
+     *
+     * @param  mixed  $data
+     * @param  int  $flags
+     * @param  int  $depth
+     * @return string
+     *
+     * @throws \JsonException
+     */
+    protected function convertDataToJavaScriptExpression($data, $flags = 0, $depth = 512)
+    {
+        $json = $this->jsonEncode($data, $flags, $depth);
+
+        if (is_string($data)) {
+            return "'".substr($json, 1, '-1')."'";
```

Comment:
3rd parameter of substr should be int (not string).

Reply:
Oof. Not sure how that slipped in there!